addons/l10n_it_edi/models/account_move.py

```diff
@@ -1168,8 +1168,8 @@ def build_error(message, records):
             errors['move_reverse_charge_with_mixed_services_and_goods'] = build_error(
                 message=_("Cannot apply Reverse Charge to bills which contains both services and goods."),
                 records=moves)
-        if pa_moves := self.filtered(lambda move: move.company_id.partner_id._l10n_it_edi_is_public_administration()):
-            if moves := pa_moves.filtered(lambda move: move.l10n_it_origin_document_type):
+        if pa_moves := self.filtered(lambda move: move.partner_id._l10n_it_edi_is_public_administration()):
```

Comment:
looking at `._l10n_it_edi_is_public_administration` in other location, it seems to be always used with `commercial_partner_id` and not `partner_id` so we should probably do that here too

```suggestion
        if pa_moves := self.filtered(lambda move: move.commercial_partner_id._l10n_it_edi_is_public_administration()):
```
commercial_partner_id is the first parent of partner_id or itself that is a company, so for example if partner is "FP" that has a parent partner "Odoo" that is a company, partner_id would be "FP" but the commercial partner would "Odoo". And if the partner_id is "Odoo" the commercial_partner_id would be "Odoo" itself.